modules/cxf/src/main/java/org/mule/module/cxf/CxfInboundMessageProcessor.java

```diff
@@ -422,12 +405,51 @@ private MuleEvent sendThroughCxf(MuleEvent event, Exchange exchange) throws Tran
 
         // invoke the actual web service up until right before we serialize the
         // response
-        d.getMessageObserver().onMessage(m);
+        try
+        {
+            d.getMessageObserver().onMessage(m);
+        }
+        catch (SuspendedInvocationException e)
+        {
+            MuleEvent responseEvent = (MuleEvent) exchange.get(CxfConstants.MULE_EVENT);
+
+            if (responseEvent == null || !responseEvent.equals(NonBlockingVoidMuleEvent.getInstance()))
```

Comment:
Why is exception swallowed if event is NonBlockingVoidMuleEvent?  This doesn't happen per-request does it? And if it doesn't happen per request, surely it's an error case and the exception needs handling, logging or propagating?